poly_commit/src/kzg/hyper_kzg_structs.rs

```diff
@@ -0,0 +1,362 @@
+use std::ops::{Index, IndexMut};
+
+use arith::ExtensionField;
+use halo2curves::{ff::Field, pairing::Engine};
+use itertools::izip;
+use transcript::Transcript;
+
+use crate::*;
+
+#[derive(Clone, Debug)]
+pub struct HyperKZGExportedLocalEvals<E: Engine> {
+    pub beta_x2_eval: E::Fr,
+    pub pos_beta_x_evals: Vec<E::Fr>,
+    pub neg_beta_x_evals: Vec<E::Fr>,
+}
+
+impl<E: Engine> Default for HyperKZGExportedLocalEvals<E> {
```

Comment:
hmmm.. I see there are a few impl here because `Engine` does not impl `Default`.

one alternative way i used to use is

```
#[derive(Clone, Debug, Derivative)]
#[derivative(Default(bound=""))]
pub(crate) struct HyperKZGLocalEvals<E: Engine> {
    pub(crate) beta2_evals: Vec<E::Fr>,
    pub(crate) pos_beta_evals: Vec<E::Fr>,
    pub(crate) neg_beta_evals: Vec<E::Fr>,
}
``` 


this requires `derivative = "2.2"` package... and it allows for bypassing the trait bound check for E on default...

// just FYI, and I don't have a preference one way or another

Reply:
sure, I will take on this one, I recall that the Engine Default thing is a quite pain in the butt